packages/react/src/hooks/use-callback-ref.ts

```diff
@@ -14,10 +14,12 @@ export function useCallbackRef<Args extends unknown[], Return>(
     throw new Error("Cannot call an event handler while rendering.")
   })
 
+  // Use useInsertionEffect to update the ref before React processes effects
   useInsertionEffect(() => {
     callbackRef.current = callback
-  })
+  }, [callback])
 
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  return useCallback((...args: Args) => callbackRef.current?.(...args), deps)
+  // We're removing the dependency array because we want this callback to be stable
+  // across renders, and only the internal ref.current should change
+  return useCallback((...args: Args) => callbackRef.current?.(...args), [])
```

Comment:
```suggestion
  return useCallback((...args: Args) => callbackRef.current?.(...args), deps)
```
It will be stable because of the default `[]` value on the line 11 `deps: React.DependencyList = [],`.
`deps` allows us to surgically update reference form the user-land which is useful in some edge-cases, but it doesn't make this callback unstable.